DQM/CTPPS/plugins/CTPPSRandomDQMSource.cc

```diff
@@ -0,0 +1,201 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+
+#include "CondFormats/PPSObjects/interface/CTPPSPixelIndices.h"
+#include "DataFormats/CTPPSDetId/interface/CTPPSDetId.h"
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigi.h"
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDataError.h"
+#include "DataFormats/CTPPSReco/interface/CTPPSPixelCluster.h"
+#include "DataFormats/CTPPSReco/interface/CTPPSPixelLocalTrack.h"
+#include "DataFormats/Common/interface/TriggerResults.h"
+
+#include <string>
+
+//-----------------------------------------------------------------------------
+
+class CTPPSRandomDQMSource : public DQMEDAnalyzer {
+public:
+  CTPPSRandomDQMSource(const edm::ParameterSet &ps);
+  ~CTPPSRandomDQMSource() override;
+
+protected:
+  void dqmBeginRun(edm::Run const &, edm::EventSetup const &) override;
+  void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
+  void analyze(edm::Event const &e, edm::EventSetup const &eSetup) override;
+
+private:
+  edm::EDGetTokenT<edm::DetSetVector<CTPPSPixelDigi>> tokenDigi;
+
+  static constexpr int NArms = 2;
+  static constexpr int NStationMAX = 3;  // in an arm
+  static constexpr int NRPotsMAX = 6;    // per station
+  static constexpr int NplaneMAX = 6;    // per RPot
+  static constexpr int RPn_first = 3, RPn_last = 4;
+  static constexpr int StationIDMAX = 4;  // possible range of ID
+  static constexpr int RPotsIDMAX = 8;    // possible range of ID
+
+  unsigned int rpStatusWord = 0x8008;      // 220_fr_hr(stn2rp3)+ 210_fr_hr
+  int RPstatus[StationIDMAX][RPotsIDMAX];  // symmetric in both arms
+  int StationStatus[StationIDMAX];         // symmetric in both arms
+  const int IndexNotValid = 0;
+
+  MonitorElement *hBX;
+
+  static constexpr int RPotsTotalNumber = NArms * NStationMAX * NRPotsMAX;
+
+  int RPindexValid[RPotsTotalNumber];
+  MonitorElement *h2HitsVsBXRandoms[RPotsTotalNumber];
+
+  int getRPindex(int arm, int station, int rp) {
+    if (arm < 0 || station < 0 || rp < 0)
+      return (IndexNotValid);
+    if (arm > 1 || station >= NStationMAX || rp >= NRPotsMAX)
+      return (IndexNotValid);
+    int rc = (arm * NStationMAX + station) * NRPotsMAX + rp;
+    return (rc);
+  }
+};
+
+//----------------------------------------------------------------------------------
+
+using namespace std;
+using namespace edm;
+
+//-------------------------------------------------------------------------------
+
+CTPPSRandomDQMSource::CTPPSRandomDQMSource(const edm::ParameterSet &ps)
+    : rpStatusWord(ps.getUntrackedParameter<unsigned int>("RPStatusWord", 0x8008)) {
+  tokenDigi = consumes<DetSetVector<CTPPSPixelDigi>>(ps.getUntrackedParameter<edm::InputTag>("tagRPixDigi"));
```

Comment:
I don't think these should be `untracked`, as they affect the outputs of the DQM plugin.

Reply:
> I don't think these should be `untracked`, as they affect the outputs of the DQM plugin.

@emanueleusai can correct me if I'm wrong, but as far as I remember there is the habit of keeping untracked the parameters in the DQM modules

Reply:
Okay, thanks. I didn't know this, so sorry for the confusion. It would be useful if someone (e.g. DQM) could clarify what rationale one should follow in this case (is it 'always untracked for DQM' ?).

Reply:
I'm leaving this pending. I will adjust it as soon as DQM clarifies.

Reply:
right now it's a mix of both. Last year there was some initial work being done to uniform things and convert all params to untracked with the reasoning being that DQM does not produce "physics" quantities hence params should not be tracked. However when we started converting all modules to untracked, we left input collections as tracked for the reason Marino mentions above. 

In short, the current policy is: input collections tracked, everything else untracked

Reply:
Alright, so I think that the InputTag should be put as tracked, if we want to follow this convention. I'm doing it and pushing another commit.